### PR TITLE
EDO-256: Support pure training mode for teams

### PR DESCRIPTION
**Pull request by fbelter-iteratec**

### CODE REVIEW

File: src/main/java/de/otto/teamdojo/domain/Team.java [hunk 1]
```diff
@@ -50,7 +50,11 @@
     @Column(name = "valid_until")
     private Instant validUntil;
 
-    @ManyToMany(fetch = FetchType.EAGER)
```
Comment (sberlin) [inline src/main/java/de/otto/teamdojo/domain/Team.java]: This fetch type had a purpose. Are you sure we can drop it?

File: src/main/webapp/app/entities/skill/skill-update.component.html [hunk 2]
```diff
@@ -76,7 +76,7 @@ <h2 dojoTranslate="teamdojoApp.skill.home.createOrEditLabel" id="jhi-skill-headi
                         </small>
                     </div>
                 </div>
-                <span>{{ 'teamdojoApp.skill.validationExplanation' | translate }}</span>
+                <span>{{ 'teamdojoApp.team.pureTrainingTeamExplanation' | translate }}</span>
```
Comment (sberlin) [inline src/main/webapp/app/entities/skill/skill-update.component.html]: Why does skill need a change for this PR?

File: src/main/webapp/app/shared/model/team.model.ts [hunk 3]
```diff
@@ -25,11 +26,14 @@ export class Team implements ITeam {
...
+        public pureTrainingTeam?: boolean,
         public participations?: IDimension[],
         public skills?: ITeamSkill[],
         public imageName?: string,
         public imageId?: number,
         public expired?: boolean,
         public daysUntilExpiration?: number
-    ) {}
+    ) {
+        this.pureTrainingTeam = this.pureTrainingTeam || false;
```
Comment (sberlin) [inline src/main/webapp/app/shared/model/team.model.ts]: This looks like a bug. Did you mean `this.pureTrainingTeam = pureTrainingTeam || false;`
Additionally the last amendment in the ticket wants this value to be true by default.

File: src/test/java/de/otto/teamdojo/test/util/TeamTestDataProvider.java [hunk 4]
```diff
@@ -35,10 +35,12 @@ public static TeamBuilder ft2() {
         private final String shortName;
         private String slogan;
         private String contactPerson;
+        private boolean pureTrainingTeam;
 
         public TeamBuilder(String name, String shortName) {
             this.name = name;
             this.shortName = shortName;
+            this.pureTrainingTeam = false;
```
Comment (sberlin) [inline src/test/java/de/otto/teamdojo/test/util/TeamTestDataProvider.java]: The last amendment in the ticket wants this value to be true by default.

**Comment by 5Heinzelmann** [comment]
Changes are made in https://github.com/iteratec/iteratec-teamdojo/pull/83

**Pull request merged**